modules/zeromq/src/test/java/org/apache/ignite/stream/zeromq/IgniteZeroMqStreamerTest.java

```diff
@@ -0,0 +1,228 @@
+import org.apache.ignite.events.CacheEvent;
+import org.apache.ignite.lang.IgnitePredicate;
+import org.apache.ignite.testframework.junits.common.GridCommonAbstractTest;
+import org.jetbrains.annotations.NotNull;
+import org.zeromq.ZMQ;
+
+import static org.apache.ignite.events.EventType.EVT_CACHE_OBJECT_PUT;
+
+/**
+ *
+ */
+public class IgniteZeroMqStreamerTest extends GridCommonAbstractTest {
+    /** Cache entries count. */
+    private static final int CACHE_ENTRY_COUNT = 1000;
+
+    /**  */
+    private final String ADDR = "tcp://localhost:5671";
+
+    /**  */
+    private final byte[] TOPIC = "0mq".getBytes();
+
+    /**  */
+    public IgniteZeroMqStreamerTest() {
+        super(true);
+    }
+
+    /** {@inheritDoc} */
+    @Override protected long getTestTimeout() {
+        return 10_000;
+    }
+
+    /** {@inheritDoc} */
+    @Override public void beforeTestsStarted() throws Exception {
+        grid().getOrCreateCache(defaultCacheConfiguration());
+    }
+
+    /** {@inheritDoc} */
+    @Override public void afterTestsStopped() throws Exception {
+        stopAllGrids();
+    }
+
+    /**
+     * @throws Exception Test exception.
+     */
+    public void testZeroMqPairSocket() throws Exception {
+        try (IgniteDataStreamer<Integer, String> dataStreamer = grid().dataStreamer(null)) {
+            try (IgniteZeroMqStreamer streamer = newStreamerInstance(
+                dataStreamer, 1, ZeroMqTypeSocket.PAIR, ADDR, null);) {
+                executeStreamer(streamer, ZMQ.PAIR, null);
+            }
+        }
+    }
+
+    /**
+     * @throws Exception Test exception.
+     */
+    public void testZeroMqSubSocket() throws Exception {
+        try (IgniteDataStreamer<Integer, String> dataStreamer = grid().dataStreamer(null)) {
+            try (IgniteZeroMqStreamer streamer = newStreamerInstance(
+                dataStreamer, 1, ZeroMqTypeSocket.SUB, ADDR, TOPIC);) {
+                executeStreamer(streamer, ZMQ.PUB, TOPIC);
+            }
+        }
+    }
+
+    /**
+     * @throws Exception Test exception.
+     */
+    public void testZeroMqPullSocket() throws Exception {
+        try (IgniteDataStreamer<Integer, String> dataStreamer = grid().dataStreamer(null)) {
+            try (IgniteZeroMqStreamer streamer = newStreamerInstance(
+                dataStreamer, 1, ZeroMqTypeSocket.PULL, ADDR, null);) {
+                executeStreamer(streamer, ZMQ.PUSH, null);
+            }
+        }
+    }
+
+    /**
+     * @param streamer ZeroMQ streamer.
+     * @param clientSocket .
+     * @param topic .
+     * @throws InterruptedException Test exception.
+     */
+    private void executeStreamer(IgniteZeroMqStreamer streamer, int clientSocket,
+        byte[] topic) throws InterruptedException {
+        streamer.setSingleTupleExtractor(new ZeroMqStringSingleTupleExtractor());
+
+        // Checking streaming.
+        CacheListener listener = subscribeToPutEvents();
+
+        streamer.start();
+
```

Comment:
Make sure the cache is empty `assertEquals(0, cache.size(CachePeekMode.PRIMARY));`